hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -514,6 +514,96 @@ object LowerTableIR {
                 StreamFilter(rows, "row", cond))
           }
 
+        case TableFilterIntervals(child, intervals, keep) =>
+          val loweredChild = lower(child)
+          val part = loweredChild.partitioner
+          val kt = child.typ.keyType
+          val ord = PartitionBoundOrdering(kt)
+          val iord = ord.intervalEndpointOrdering
+          val nPartitions = part.numPartitions
+
+
+          val filterPartitioner = new RVDPartitioner(kt, Interval.union(intervals.toArray, ord.intervalEndpointOrdering), 0)
```

Comment:
I don't think you want `allowedOverlap = 0`. That's the strongest restriction, so if `intervals` is `[[{0, 1}, {2, 5}], [{2, 7}, {3, 0}]]`, this would throw an error because the key prefix `2` is split across partitions. I think you want no restriction here, which means you can just leave out `allowedOverlap` (the default is `kt.size`).

Reply:
Interval.union returns a sorted array of intervals that have this property, this should be OK

Reply:
That doesn't sound right. In my example
```
[[{0, 1}, {2, 5}], [{2, 7}, {3, 0}]]
```
union wouldn't change anything, because the two intervals don't overlap, right? But the prefix `{2}` does overlap, so the partitioner doesn't satisfy `allowedOverlap = 0`. I think the result of union will always satisfy `allowedOverlap = kt.size - 1` (equivalently, `partitionKey = kt.fieldNames`), if you want to enforce that.

Reply:
ah... I see. Thanks for explaining.